File: module_wan/src/test/java/com/huyingbao/module/wan/module/MockComponent.java
package com.huyingbao.module.wan.module;

import com.huyingbao.module.wan.action.WanApi;

import javax.inject.Singleton;

import dagger.Component;

/**
 * @author liujunfeng
 * @date 2019/1/1
 */
@Singleton
@Component(modules = {MockModule.class})
public interface MockComponent {
    WanApi getWanApi();
}

File: module-wan/src/test/java/com/huyingbao/module/wan/module/MockComponent.kt
package com.huyingbao.module.wan.module

import com.huyingbao.module.wan.api.WanApi

import javax.inject.Singleton

import dagger.Component

/**
 * mock注入器,为测试代码提供,方便测试的全局对象
 *
 *
 * Created by liujunfeng on 2019/1/1.
 */
@Singleton
@Component(modules = [MockModule::class])
interface MockComponent {
    val wanApi: WanApi
}
